src/UCTSearch.cpp

```diff
@@ -247,7 +247,7 @@ bool UCTSearch::should_resign(passflag_t passflag, float bestscore) {
     }
 
     const auto visits = m_root->get_visits();
-    if (visits < std::min(500, cfg_max_playouts))  {
+    if (visits < std::min(500, std::min(cfg_max_visits, cfg_max_playouts)))  {
```

Comment:
It might be more elegant to make it a single `std::min` taking an initializer_list of three arguments:
`std::min({500, cfg_max_visits, cfg_max_playouts})`
This does come with the unfortunate gotcha that both GCC and Clang will generate unhelpful error messages if`<algorithm>` isn't included now.

Reply:
Sounds like cleaner code to me, so I like your suggestion. Just add <algorithm> to the includes.

Reply:
Done!